core/trino-main/src/main/java/io/trino/execution/DropTableTask.java

```diff
@@ -76,7 +78,17 @@ public ListenableFuture<Void> execute(
                     "Table '%s' does not exist, but a view with that name exists. Did you mean DROP VIEW %s?", originalTableName, originalTableName);
         }
 
-        RedirectionAwareTableHandle redirectionAwareTableHandle = metadata.getRedirectionAwareTableHandle(session, originalTableName);
+        RedirectionAwareTableHandle redirectionAwareTableHandle;
+        try {
+            redirectionAwareTableHandle = metadata.getRedirectionAwareTableHandle(session, originalTableName);
+        }
+        catch (TrinoException e) {
+            if (e.getErrorCode().equals(METADATA_NOT_FOUND.toErrorCode())) {
```

Comment:
So we're relying on the connector throwing this error code for the case of "a table exists, but it's corrupt". 

Do we need it? Maybe we just try to delete by name regardless of the error type

Reply:
Yes, we don't need the error code check here. Removed the check.